java-manta-client-unshaded/src/test/java/com/joyent/manta/client/MantaMetadataTest.java

```diff
@@ -64,14 +68,37 @@ public void cantAddMetadataKeyThatIllegalCharacter() {
 
             try {
                 instance.put(key, "world");
-            } catch (IllegalArgumentException e) {
+            } catch (final IllegalArgumentException e) {
                 caught = true;
             }
 
             Assert.assertTrue(caught, String.format("No exception thrown for char: %s", c));
         }
     }
 
+    // Until MANTA-3527 is resolved we should prevent users from using non-ascii values for both keys
```

Comment:
Even if MANTA-3527 is resolved, we will want to prevent the users from using non-ascii values in the headers / metadata.

Reply:
https://tools.ietf.org/html/rfc5987 seems to imply that ISO-8859-1 is the accepted character set but HTTP/1.1 suggests sticking to US-ASCII in the final paragraph of [this section](https://tools.ietf.org/html/rfc7230#section-3.2.4).

Since users can construct and manipulate their own `MantaHttpHeaders` objects does this mean we want to extend the validation to that class as well (i.e. use `HttpHeaderPredicate` in both classes)?